analyzer.go

```diff
@@ -78,16 +78,28 @@ func (a *Analyzer) analyzeLayers(appMeta platform.LayersMetadata, cache Cache) e
 		// Restore metadata for launch=true layers.
 		// The restorer step will restore the layer data for cache=true layers if possible or delete the layer.
 		appLayers := appMeta.MetadataForBuildpack(buildpack.ID).Layers
+		cachedLayers := cacheMeta.MetadataForBuildpack(buildpack.ID).Layers
 		for name, layer := range appLayers {
 			identifier := fmt.Sprintf("%s:%s", buildpack.ID, name)
 			if !layer.Launch {
 				a.Logger.Debugf("Not restoring metadata for %q, marked as launch=false", identifier)
 				continue
 			}
 			if layer.Build && !layer.Cache {
+				// layer is launch=true, build=true. Because build=true, the layer contents must be present in the build container.
+				// There is no reason to restore the metadata file, because the buildpack will always recreate the layer.
 				a.Logger.Debugf("Not restoring metadata for %q, marked as build=true, cache=false", identifier)
 				continue
 			}
+			if layer.Cache {
+				if cacheLayer, ok := cachedLayers[name]; !ok || !cacheLayer.Cache {
+					// The layer is not cache=true in the cache metadata and will not be restored.
+					// Do not write the metadata file so that it is clear to the buildpack that it needs to recreate the layer.
+					// Although a launch=true (only) layer still needs a metadata file, the restorer will remove the file anyway when it does its cleanup (for buildpack apis < 0.6).
```

Comment:
@ekcasey @jabrown85, I think that the restorer wasn't supposed to delete the file in this case but since it does, we wanted to be consistent.